### PR TITLE
Make encryption HMAC digest configurable

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Make `HMAC digest` configurable through `Config\Encryption` or through `.env` file. Also optimized some code in the Encryption library.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPdocs
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by michalsn** [comment]
Well... we could add `cipher` as well, but... Personally, I have a little problem with that. Because every time we make these configurable things user-friendly... people without proper knowledge show up and make changes that they shouldn't be doing ;)

So... basically I think that "hiding" these config variables was intentional.

File: system/Encryption/Encryption.php [hunk 1]
```diff
@@ -132,10 +131,11 @@ public function __construct(BaseConfig $config = null)
 	public function initialize(BaseConfig $config = null)
 	{
 		// override config?
-		if (! empty($config))
+		if ($config)
 		{
-			$this->driver = $config->driver;
 			$this->key    = $config->key;
+			$this->driver = $config->driver;
+			$this->digest = $config->digest;
```
Comment (samsonasik) [inline system/Encryption/Encryption.php:138]: probably add default value:

```suggestion
                      $this->digest = $config->digest ?? 'SHA512';
```

File: system/Encryption/Encryption.php [hunk 2]
```diff
@@ -99,12 +99,11 @@ class Encryption
...
-		if (empty($config))
-		{
-			$config = new \Config\Encryption();
-		}
-		$this->driver = $config->driver;
+		$config = $config ?? new \Config\Encryption();
+
 		$this->key    = $config->key;
+		$this->driver = $config->driver;
+		$this->digest = $config->digest;
```
Comment (samsonasik) [inline system/Encryption/Encryption.php:106]: probably add default value:

```suggestion
              $this->digest = $config->digest ?? 'SHA512';
```

**Comment by paulbalandan** [comment]
> Well... we could add `cipher` as well, but... Personally, I have a little problem with that. Because every time we make these configurable things user-friendly... people without proper knowledge show up and make changes that they shouldn't be doing ;)
> 
> So... basically I think that "hiding" these config variables was intentional.

I totally agree with your point here. It is quite annoying someone would mess the configs and blaming the framework for it as a bug. Adding `cipher` as well is a bit too much, I think. My concern here is like in #3404 where hosting providers don't provide databases that handle big encrypted messages nor modern charset handling.

**Comment by paulbalandan** [comment]
@MGatner, yes this is ready.

**Pull request merged**